blockstore/api.go

```diff
@@ -15,14 +16,22 @@ type ChainIO interface {
 }
 
 type apiBlockstore struct {
-	api ChainIO
+	api   ChainIO
+	cache *lru.Cache[cid.Cid, []byte]
 }
 
 // This blockstore is adapted in the constructor.
 var _ BasicBlockstore = (*apiBlockstore)(nil)
 
 func NewAPIBlockstore(cio ChainIO) Blockstore {
-	bs := &apiBlockstore{api: cio}
+	lc, err := lru.New[cid.Cid, []byte](1024)
+	if err != nil {
+		panic(err)
```

Comment:
Subjective: I would leave a comment about why this panic. It's cheap to do and reduces cognitive cycles for future readers: "`lru.New` only returns an error the size is less than 1. There is no point returning this error since the size is configured as a constant."